zk/conn.go

```diff
@@ -475,6 +487,17 @@ func (c *Conn) loop() {
 		}
 		c.flushRequests(err)
 
+		if err == ErrSessionExpired && !c.reconnectOnSessionExpiration {
+			atomic.StoreInt32(&c.sessionExpired, 1)
+
+			// The process of flushing requests is racy by design so do it in two steps.
+			c.flushUnsentRequests(err)
```

Comment:
This diff adds dropping of requests before trying to insert them into `sendChan`, check `(*Conn) request)` method below.

Re: Close. Currently `Close` is leaky (not sure if I should address it here or in another diff):
```
zc.Close()
zc.Get(somePath) // <- this blocks forever as it tries to put data into zc.sendChan and nobody reads it.
```